src/v/pandaproxy/error.cc

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "error.h"
+
+#include "kafka/protocol/errors.h"
+#include "pandaproxy/parsing/error.h"
+
+namespace pandaproxy {
+
+namespace {
+
+struct reply_error_category final : std::error_category {
+    const char* name() const noexcept override { return "pandaproxy"; }
+    std::string message(int ev) const override {
+        switch (static_cast<reply_error_code>(ev)) {
+        case reply_error_code::kafka_bad_request:
+            return "kafka_bad_request";
+        case reply_error_code::kafka_authentication_error:
+            return "kafka_authentication_error";
+        case reply_error_code::kafka_authorization_error:
+            return "kafka_authorization_error";
+        case reply_error_code::topic_not_found:
+            return "topic_not_found";
+        case reply_error_code::partition_not_found:
+            return "partition_not_found";
+        case reply_error_code::consumer_instance_not_found:
+            return "consumer_instance_not_found";
+        case reply_error_code::zookeeper_error:
+            return "zookeeper_error";
+        case reply_error_code::kafka_error:
+            return "kafka_error";
+        case reply_error_code::kafka_retriable_error:
+            return "kafka_retriable_error";
+        case reply_error_code::ssl_unavailable:
+            return "ssl_unavailable";
+        case reply_error_code::broker_not_available:
+            return "broker_not_available";
+        }
+        return "(unrecognized error)";
+    }
+};
+
+const reply_error_category reply_error_category{};
+
+struct kafka_error_category final : std::error_category {
```

Comment:
absolute :fire:

Reply:
It's 100% correct with an unknown confidence interval. A lot of it was guesswork.